### PR TITLE
Remove scroll search for ES7 and new pagination

### PR DESCRIPTION
**Pull request by higs4281**
This change is the last step in moving the consumer complaint search app to Elasticsearch 7. 
Two related code bases – the search front end at [ccdb5-iu](https://github.com/cfpb/ccdb5-ui) and the indexing pipeline at [ccdb-data-pipeline](https://github.com/cfpb/ccdb-data-pipeline) – have been updated.

There are a lot of file changes here, but the bulk of them are test fixtures and tests that needed to be tweaked.

The main changes are in two files: 
- complaint_search/es_interface.py
- complaint_search/es_builders.py

And there are a few new tests for [search_after](https://github.com/cfpb/ccdb5-api/blob/4b2e9fcc9dfbd1542e258fec2004d21443cee20a/complaint_search/tests/test_es_interface.py#L26-L107) and [break_points](https://github.com/cfpb/ccdb5-api/blob/4b2e9fcc9dfbd1542e258fec2004d21443cee20a/complaint_search/tests/test_get_break_points.py).

## Changes
Elasticsearch no longer supports using a "from" value to paginate results of a
"scroll" search, so our search back end needed to be refactored to restore deep pagination.
This PR implements Elasticsearch's recommended use of "search_after" to paginate results.

We also needed to cope with changes to search structure and changes in how search count and aggregation totals are obtained from the Elasticsearch API.

## Testing
This can be tested locally by following the [INSTALL](https://github.com/cfpb/ccdb-data-pipeline/blob/update-installation-docs/INSTALL.md) directions in the ccdb-data-pipeline repo and
getting a local index populated, or by checking out our DEV5 server, where this code is running.

~~Draft for now, since this has no related front-end handling yet, and there's some big testing reconciliation to do.
The main added function, complaint_search.es_interface.get_break_points, has tests.~~

Tests are running, pagination is working, and this is ready for review.

### CODE REVIEW

File: complaint_search/tests/test_es_interface.py [hunk 1]
```diff
@@ -19,20 +19,27 @@
...
+        self.maxDiff = None
+
     # -------------------------------------------------------------------------
     # Helper Attributes
     # -------------------------------------------------------------------------
     MOCK_SEARCH_SIDE_EFFECT = [
         {
             "search": "OK",
-            "_scroll_id": "This_is_a_scroll_id",
+            # "_scroll_id": "This_is_a_scroll_id",
```
Comment (lorenseanstewart) [inline complaint_search/tests/test_es_interface.py:36]: did you mean to leave this here?

File: complaint_search/tests/test_es_interface.py [hunk 2]
```diff
@@ -53,25 +60,31 @@ class EsInterfaceTest_Search(TestCase):
         }
     ]
 
-    MOCK_COUNT_RETURN_VALUE = {"count": 100}
+    MOCK_COUNT_RETURN_VALUE = {"count": 4}
 
     MOCK_SEARCH_RESULT = {
         'search': 'OK',
-        "_scroll_id": "This_is_a_scroll_id",
+        # "_scroll_id": "This_is_a_scroll_id",
```
Comment (lorenseanstewart) [inline complaint_search/tests/test_es_interface.py:67]: same as above

File: complaint_search/views.py [hunk 3]
```diff
@@ -10,6 +10,8 @@
     EXCLUDE_PREFIX,
     EXPORT_FORMATS,
     FORMAT_CONTENT_TYPE_MAP,
+    # RESULT_SIZE_DEFAULT,
+    # RESULT_SIZE_OPTIONS,
```
Comment (lorenseanstewart) [inline complaint_search/views.py:14]: same as above

**Review by lorenseanstewart — APPROVED** [review]
From what I can wrap my head around, the code looks good. I left several comments about commented out code, but feel free to ignore them if you have a reason to leave the code there.

Do you want me to spin this up locally to make sure it works?

File: complaint_search/tests/test_es_interface.py (hunk 1 above)
Comment (higs4281) [reply complaint_search/tests/test_es_interface.py:36]: Good catch.

File: complaint_search/tests/test_es_interface.py (hunk 2 above)
Comment (higs4281) [reply complaint_search/tests/test_es_interface.py:67]: √

**Comment by higs4281** [comment]
> Do you want me to spin this up locally to make sure it works?

Only if you want the exercise. The code is running in DEV5 and can be test-driven there more easily.

**Pull request merged**